File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/roleplay/quest/QuestActiveDetailedInformations.scala





















// Generated on 05/03/2013 10:35:05
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait QuestActiveDetailedInformations extends QuestActiveInformations {
  override val typeId = 382

  protected trait Type extends QuestActiveInformations.Type with Serializable {
    override val typeId = QuestActiveDetailedInformations.typeId

    val stepId: Short
    val objectives: Seq[QuestObjectiveInformations.TypeImpl]

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeShort(stepId)
      buf.writeUShort(objectives.length)
      for (entry <- objectives) {
        buf.writeUShort(entry.typeId)
        entry.serialize(buf)
      }
    }

  }

  protected def create(questId: Short, stepId: Short, objectives: Seq[QuestObjectiveInformations.TypeImpl]): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val questId = buf.readShort()
    val stepId = buf.readShort()
    val objectives = {
      val builder = collection.immutable.Seq.newBuilder[QuestObjectiveInformations.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += types.get[QuestObjectiveInformations.TypeImpl](buf.readShort());
      }
      builder.result()
    }

    create(questId, stepId, objectives)
  }

}

object QuestActiveDetailedInformations extends QuestActiveDetailedInformations {
  case class TypeImpl(questId: Short, stepId: Short, objectives: Seq[QuestObjectiveInformations.TypeImpl]) extends Type
  override def create(questId: Short, stepId: Short, objectives: Seq[QuestObjectiveInformations.TypeImpl]) = TypeImpl(questId, stepId, objectives)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(questId: Short, stepId: Short, objectives: Seq[QuestObjectiveInformations.TypeImpl]) = create(questId, stepId, objectives)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/QuestActiveDetailedInformations.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class QuestActiveDetailedInformations extends QuestActiveInformations {
    public static final short TYPE_ID = 382;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short stepId;
    public QuestObjectiveInformations[] objectives;
    
    public QuestActiveDetailedInformations() { }
    
    public QuestActiveDetailedInformations(short questId, short stepId, QuestObjectiveInformations[] objectives) {
        super(questId);
        this.stepId = stepId;
        this.objectives = objectives;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeShort(stepId);
        buf.writeUShort(objectives.length);
        for (QuestObjectiveInformations entry : objectives) {
            buf.writeShort(entry.getTypeId());
            entry.serialize(buf);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        stepId = buf.readShort();
        if (stepId < 0)
            throw new RuntimeException("Forbidden value on stepId = " + stepId + ", it doesn't respect the following condition : stepId < 0");
        int limit = buf.readUShort();
        objectives = new QuestObjectiveInformations[limit];
        for (int i = 0; i < limit; i++) {
            objectives[i] = ProtocolTypeManager.getInstance().build(buf.readShort());
            objectives[i].deserialize(buf);
        }
    }
    
}
